dev/set_matrix.py

```diff
@@ -379,7 +451,13 @@ def expand_config(config):
             "sentence-transformers",
             "torch",
         )
+        validate_test_coverage(name, cfgs)
         for category, cfg in cfgs.items():
+            if category not in VALID_CATEGORIES:
```

Comment:
Can we validate this using pydantic models by disallowing extra fields?

Reply:
Ah ok so you mean adding a new Pydantic Model to represent each flavor config. Sure I'll do so.

Reply:
Yeah exactly :)

Reply:
@B-Step62 If that's a big change, feel free to file a follow-up PR.

Reply:
Not a big deal, will add it in this PR:)